packages/mobile/src/fiatExchanges/FiatExchangeAmount.tsx

```diff
@@ -0,0 +1,193 @@
+
+  const parsedInputAmount = parseInputAmount(inputAmount, decimalSeparator)
+  const dollarAmount = convertDollarsToMaxSupportedPrecision(
+    (!parsedInputAmount.isNaN() &&
+      convertLocalAmountToDollars(parsedInputAmount, localCurrencyCode ? localExchangeRate : 1)) ||
+      new BigNumber('0')
+  )
+
+  return (
+    <SafeAreaView
+      // Force inset as this screen uses auto focus and KeyboardSpacer padding is initially
+      // incorrect because of that
+      forceInset={{ top: 'never', bottom: 'always' }}
+      style={styles.container}
+    >
+      <DisconnectBanner />
+      <KeyboardAwareScrollView
+        keyboardShouldPersistTaps={'always'}
+        contentContainerStyle={styles.contentContainer}
+      >
+        <View style={styles.amountInputContainer}>
+          <View>
+            <Text style={styles.exchangeBodyText}>{t('global:amount')}</Text>
+          </View>
+          <TextInput
+            autoFocus={true}
+            keyboardType={'decimal-pad'}
+            onChangeText={onChangeExchangeAmount}
+            value={inputAmount}
+            placeholderTextColor={colors.gray3}
+            placeholder={'0'}
+            style={styles.currencyInput}
+            testID="FiatExchangeInput"
+          />
+        </View>
+        <LineItemRow
+          textStyle={styles.subtotalBodyText}
+          title={
+            <Trans i18nKey="celoDollarsAt" ns={Namespaces.fiatExchangeFlow}>
+              Celo Dollars @ <CurrencyDisplay amount={oneDollarAmount} />
+            </Trans>
+          }
+          amount={
+            <CurrencyDisplay
+              amount={{ value: dollarAmount, currencyCode: CURRENCIES[CURRENCY_ENUM.DOLLAR].code }}
+              hideSymbol={true}
+              showLocalAmount={false}
+            />
+          }
+        />
+      </KeyboardAwareScrollView>
+      <Text style={styles.dislamerCeloDollars}>{t('dislamerCeloDollars')}</Text>
+      <Button
+        onPress={goNext}
+        text={t('global:next')}
+        accessibilityLabel={t('global:next')}
+        disabled={!isNextButtonValid()}
+        size={BtnSizes.FULL}
+        style={styles.reviewBtn}
+        testID="FiatExchangeNextButton"
+      />
+      <KeyboardSpacer />
+    </SafeAreaView>
+  )
+}
+
+export default ExchangeTradeScreen
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    justifyContent: 'space-between',
+  },
+  contentContainer: {
+    paddingHorizontal: 16,
+  },
+  amountInputContainer: {
+    flexDirection: 'row',
+    marginTop: 24,
+    alignItems: 'center',
+    marginBottom: 8,
+  },
+  exchangeBodyText: {
+    ...fontStyles.regular500,
+    fontSize: 15,
+    lineHeight: 20,
+    fontWeight: '700',
+  },
+  subtotalBodyText: {
+    ...fontStyles.small,
+  },
+  currencyInput: {
+    ...fontStyles.regular,
+    marginLeft: 10,
+    flex: 1,
+    textAlign: 'right',
+    fontSize: 19,
+    lineHeight: Platform.select({ android: 27, ios: 23 }), // vertical align = center
```

Comment:
have you check this on different ios and different android phones? seems like something that could be more about a specific phone size  than platform.

Reply:
I have not checked more than two phones (real one and emulator). We have the same logic at `ExchangeTrade`, here the height was changed, therefor I've adjusted lineHeight as well. To my understanding it depends on the height of the input, which is hardcoded a couple lines later, not the screen size. But I might be wrong 🙂